zenoh-jni/src/key_expr.rs

```diff
@@ -152,3 +115,74 @@ pub(crate) unsafe extern "C" fn Java_io_zenoh_jni_JNIKeyExpr_freePtrViaJNI(
 ) {
     Arc::from_raw(ptr);
 }
+
+pub(crate) fn decode_key_expr(env: &mut JNIEnv, key_expr: &JString) -> Result<KeyExpr<'static>> {
+    let key_expr_str = decode_string(env, key_expr).map_err(|err| {
+        Error::Jni(format!(
+            "Unable to get key expression string value: {}",
+            err
+        ))
+    })?;
+    let key_expr = KeyExpr::try_from(key_expr_str).map_err(|err| {
+        Error::Jni(format!(
+            "Unable to create key expression from string: {}",
+            err
+        ))
+    })?;
+    Ok(key_expr)
+}
```

Comment:
nit: Could potentially be written a bit shorter and save on an alloc, 
not sure if this will have any real world performance change, 
but reviewer has not tried, just a nit, so up to OP to decide.